### PR TITLE
Adds Receipt Component to Money Request confirmation page

### PR DESCRIPTION
**Pull request by cdanwards**
### Details

This PR adds the `ReceiptEmptyState` component to the `MoneyRequestConfirmationList` for money requests that are associated with workspace policies of the type `Collect` or `Control`.

Note: I added the functionality to both the original file as well as the temporary conversion file to maintain consistency.
### Fixed Issues

$https://github.com/Expensify/App/issues/27744
PROPOSAL:

## Testing Instructions

**Pre-testing instructions:**

- Create a Workspace within Old Dot that is either a `Collect` or `Control` type. Instructions [here](https://expensify.slack.com/archives/C05EQBQG6E5/p1695182901255769)

---

**Expected flow for Money Request within Paid tier (Collect or Control) Workspaces:**

- [ ]  Navigate to the paid tier workspace (Select it within the LHN)
- [ ]  Within the report view, select the action button and select “Request Money”
- [ ]  Enter an amount for the money request
- [ ]  Select “Next”
- [ ]  Verify that you are on the confirmation screen
- [ ]  Verify that the `ReceiptEmptyState` component is shown (image below)

Screenshot 2024-01-11 at 12 02 11 PM

- [ ]  Press the `ReceiptEmptyState` component
- [ ]  Verify that you have been navigated to the drag and drop screen
- [ ]  Upload an image by dragging an image into the drag and drop view
- [ ]  Verify that you are navigated back to the confirmation screen
- [ ]  Verify that the correct image is displayed on the confirm screen

---

**Expected flow for Money Request *not* within Paid tier (Collect or Control) Workspaces:**

- [ ]  Navigate to a non-paid tier workspace
- [ ]  Within the report view, select the action button and select “Request Money”
- [ ]  Enter an amount for the money request
- [ ]  Select “Next”
- [ ]  Verify that you are on the confirmation screen
- [ ]  Verify that the `ReceiptEmptyState` component is *not* shown (image below)

Screenshot 2024-01-11 at 12 02 11 PM

### Offline tests

Same as Tests.

### QA Steps

Same as Tests.

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/8878532/8b40999c-8b66-4479-9a71-6aeb5e971386

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/8878532/1875b41a-5d68-439a-9537-cb4bfb07b648

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/8878532/bc07bd04-8567-4751-8df0-7f26ab155286

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/8878532/b8ed40bb-9526-4576-a82d-f70b10d75a9c

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/8878532/8bafd01d-f907-4537-b686-01ddebfe70f2

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/8878532/a1bd9858-1450-4bac-80d8-5ab457970dd6

</details>

### CODE REVIEW

**Comment by cdanwards** [comment]
I noticed that some of the uploaded videos accidentally contained personal info. Redoing those videos.

File: src/components/MoneyRequestConfirmationList.js [hunk 1]
```diff
@@ -613,6 +617,22 @@ function MoneyRequestConfirmationList(props) {
...
+                                    CONST.IOU.ACTION.CREATE,
+                                    CONST.IOU.TYPE.REQUEST,
+                                    transaction.transactionID,
+                                    props.reportID,
+                                    Navigation.getActiveRouteWithoutParams(),
+                                ),
+                            )
+                        }
+                    />
+                )
```
[the remark this answers is not in the document]
Comment (cdanwards) [reply src/components/MoneyRequestConfirmationList.js:635]: Do you mind elaborating some? This issue specifically is for the manual flow.

[the remark this answers is not in the document]
Comment (cdanwards) [reply src/components/MoneyRequestConfirmationList.js:635]: >  This component is being deprecated

Is this in reference to `MoneyRequestConfirmationList`? What do you mean by "This component"? I included the logic in component as well as the matching temporary component to maintain parity.

[the remark this answers is not in the document]
Comment (cdanwards) [reply src/components/MoneyRequestConfirmationList.js:635]: Okay! Sorry for the confusion.

**Comment by cdanwards** [comment]
Example screen recordings have been updated!

**Comment by cdanwards** [comment]
@aimane-chnaif let me know if I can help get this reviewed.

**Review by aimane-chnaif — CHANGES_REQUESTED** [review]

File: src/components/MoneyTemporaryForRefactorRequestConfirmationList.js [hunk 2]
```diff
@@ -647,6 +651,22 @@ function MoneyTemporaryForRefactorRequestConfirmationList({
                     // So if we have a thumbnail, it means we're retrieving the image from the server
                     isAuthTokenRequired={!_.isEmpty(receiptThumbnail)}
                 />
+            ) : (
+                isFromPaidPolicy && (
```
[the remark this answers is not in the document]
Comment (cdanwards) [reply src/components/MoneyTemporaryForRefactorRequestConfirmationList.js:655]: Good catch!

File: src/components/MoneyRequestConfirmationList.js [hunk 3]
```diff
@@ -222,6 +223,9 @@ function MoneyRequestConfirmationList(props) {
     const isSplitBill = props.iouType === CONST.IOU.TYPE.SPLIT;
     const isTypeSend = props.iouType === CONST.IOU.TYPE.SEND;
 
+    // A flag for checking if the associated policy is of type Team or Corporate ("Control" or "Collect")
+    const isFromPaidPolicy = props.policy && (props.policy.type === CONST.POLICY.TYPE.TEAM || props.policy.type === CONST.POLICY.TYPE.CORPORATE);
```
[the remark this answers is not in the document]
Comment (cdanwards) [reply src/components/MoneyRequestConfirmationList.js:227]: Yes! We can. I wasn't aware of this util. I'll do a quick change.

[the remark this answers is not in the document]
Comment (cdanwards) [reply src/components/MoneyRequestConfirmationList.js:227]: Also fixing the lint errors

**Comment by cdanwards** [comment]
@aimane-chnaif I tested on web and it behaved as expected. No longer showing on distance requests or splits and continues to only show on paid policy workspaces.

**Review by aimane-chnaif — APPROVED** [review]
LGTM 🎉

**Review by Julesssss — APPROVED** [review]

**Pull request merged**